### PR TITLE
node driver registrar should use CSI connection lib

### PR DESCRIPTION
**Pull request by andrewsykim**
re: https://github.com/kubernetes-csi/csi-lib-utils/issues/10 

Mostly followed https://github.com/kubernetes-csi/external-attacher/pull/123 with the exception of adding the reconnect callback on connection loss. Left it out because it doesn't handle address starting with `unix://` so the unit tests would fail:
```
--- FAIL: TestGetNodeID (0.00s)
    connection_test.go:90: OnConnectionLoss callback only supported for unix:// addresses
```
Left that option out for now. If it's important I can refactor the tests to use a different address? 

cc @msau42 @jsafrane

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 [[constraint]]
   name = "github.com/kubernetes-csi/csi-lib-utils"
-  version = "0.1.0"
+  version = "0.3.0"
```
Comment (msau42) [inline Gopkg.toml]: do we want to constrain the version here or just use the latest?

File: pkg/connection/connection.go [hunk 2]
```diff
@@ -66,38 +63,6 @@ func NewConnection(
...
-		}
-		if conn.GetState() == connectivity.Ready {
-			klog.V(3).Infof("Connected")
-			return conn, nil
-		}
-		klog.V(4).Infof("Still trying, connection is %s", conn.GetState())
-	}
-}
-
 func (c *csiConnection) GetDriverName(ctx context.Context) (string, error) {
```
Comment (msau42) [inline pkg/connection/connection.go]: @jsafrane a lot of these rpc wrappers are also shared between sidecars. Do you think we should extract these out into csi-lib-utils too?

**Comment by msau42** [comment]
> with the exception of adding the reconnect callback on connection loss

@jsafrane @pohly iiuc, the connection library will be default automatically retry connecting if the connection is lost.  So I think it's fine for now to not have any special reconnect handling for node-driver-registrar.  If we want to cache something in the future we could reconsider it.

File: pkg/connection/connection.go (hunk 2 above)
Comment (jsafrane) [reply pkg/connection/connection.go]: I'd clean up the wrappers first, is seems to me they are copied but only GetDriverName is really used.

File: cmd/csi-node-driver-registrar/main.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ const (
 
 // Command line flags
 var (
-	connectionTimeout       = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
```
Comment (mowangdk) [inline cmd/csi-node-driver-registrar/main.go:46]: Is there anyone knows why disable timeout here?

**Pull request merged**